services/dialog/v1.js

```diff
@@ -183,113 +183,114 @@ module.exports = function (RED) {
-		
+    node.status({fill:'blue', shape:'dot', text:'requesting create of new dialog template'});	 
+    //if ('file' in msg.dialog_params && 'dialog_name' in msg.dialog_params) {
+    if ('dialog_name' in msg.dialog_params) {  
       // extension supported : only XML
-      temp.open({suffix: ".xml"}, function (err, info) {
+      // TODO : support other API format (json and encrypted crt)
+      temp.open({suffix: '.xml'}, function (err, info) {
         if (err) throw err;
-        var fileBuffer = msg.dialog_params["file"];
+        //var fileBuffer = msg.dialog_params['file'];
+        var fileBuffer = msg.payload;
         fs.writeFile(info.path, fileBuffer, function (err) {
             if (err) {
               console.error(err);
               throw err;
             }
             var aFileStream = fs.createReadStream(info.path);
             params.file = aFileStream;
-            params.name = msg.dialog_params["dialog_name"];
+            params.name = msg.dialog_params['dialog_name'];
             // Watson SDK call
             dialog.createDialog(params, function(err, dialog_data){
+                  node.status({});
                   if (err) {
-                    node.status({fill:"red", shape:"ring", text:"call to dialog service failed"}); 
+                    node.status({fill:'red', shape:'ring', text:'call to dialog service failed'}); 
                     node.error(err, msg);
                   } else {
-                    node.status({fill:"green", shape:"dot", text:"Dialog template created successfully"});      
                     msg.dialog = dialog_data;     
-                    msg.payload = "Check msg.dialog dialog data";
+                    msg.payload = 'Check msg.dialog dialog data';
                     node.send(msg);
                   }       
             });  
         });
      });
-    } else if (! 'file' in msg.dialog_params) {
-      var errtxt = "Missing Dialog template file";
-      node.status({fill:"red", shape:"ring", text:errtxt}); 		
-      node.error(errtxt, msg); 
-    }  else {
-      errtxt = "Dialog Name not specified";
-      node.status({fill:"red", shape:"ring", text:errtxt}); 		
+    } /* else if (! 'file' in msg.dialog_params) {
+      var errtxt = 'Missing Dialog template file';
+      node.status({fill:'red', shape:'ring', text:errtxt}); 		
+      node.error(errtxt, msg);
+    }*/
+    else {
+      errtxt = 'dialog_name not specified';
+      node.status({fill:'red', shape:'ring', text:errtxt}); 		
       node.error(errtxt, msg); 
     }   
   }
 
 // This function delete an existing dialog instance.
   function performDelete(node,dialog,msg, config) {
     var params = {};
-    var message = "";
-    node.status({fill:"blue", shape:"dot", text:"requesting delete of a dialog instance"}); 
+    var message = '';
+    node.status({fill:'blue', shape:'dot', text:'requesting delete of a dialog instance'}); 
 
-    if (!config.dialog || "" != config.dialog) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '!==' and instead saw '!='.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2462016486/issues/source?bid=3225115&fileBranchId=3255594#l235)